lib/ROM/RomParser.cpp

```diff
@@ -0,0 +1,95 @@
+// ****************************************************************************** //
+//   _______   __                              __       __                        //
+//  /       \ /  |                            /  \     /  |                       //
+//  $$$$$$$  |$$ |  ______   __    __         $$  \   /$$ |  ______   _______     //
+//  $$ |__$$ |$$ | /      \ /  |  /  | ______ $$$  \ /$$$ | /      \ /       \    //
+//  $$    $$/ $$ | $$$$$$  |$$ |  $$ |/      |$$$$  /$$$$ | $$$$$$  |$$$$$$$  |   //
+//  $$$$$$$/  $$ | /    $$ |$$ |  $$ |$$$$$$/ $$ $$ $$/$$ | /    $$ |$$ |  $$ |   //
+//  $$ |      $$ |/$$$$$$$ |$$ \__$$ |        $$ |$$$/ $$ |/$$$$$$$ |$$ |  $$ |   //
+//  $$ |      $$ |$$    $$ |$$    $$ |        $$ | $/  $$ |$$    $$ |$$ |  $$ |   //
+//  $$/       $$/  $$$$$$$/  $$$$$$$ |        $$/      $$/  $$$$$$$/ $$/   $$/    //
+//                          /  \__$$ |                                            //
+//                          $$    $$/                                             //
+//                           $$$$$$/                                              //
+//                                                                                //
+//                            By: K1ngmar and rvan-mee                            //
+// ****************************************************************************** //
+
+#include <play-man/ROM/Rom.hpp>
+#include <play-man/utility/UtilFunc.hpp>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <cstring>
+#include <assert.h>
+
+namespace GameBoy {
+
+    void RomHeader::ParseRawData(const std::vector<int8_t>& data)
+    {
+        assert(data.size() >= 0x14E);
+        title.fill('\0');
+        manufacturerCode.fill('\0');
+
+        // Nintendo Logo
+        for (uint32_t i = 0; i < nintendoLogoSize; i++)
+            nintendoLogo[i] = data[i + nintendoLogoIndex];
+
+        // Cartridge Title
+        for (uint32_t i = 0; i < romTitleSize; i++)
+        {
+            if (!(data[i + titleIndex] >= 32 && data[i + titleIndex] <= 126))
+                break ;
+            title[i] = data[i + titleIndex];
+        }
+
+        // Manufacturer Code
+        // Not all cartridges contained a manufacturer code, instead using the space for the title
+        // TODO: figure out if there is a way to separate the title and the manufacturer code 
```

Comment:
Create issue for todo

Reply:
also link the issue code in the todo comment so that if we ever have a problem with this we know which  issue is linked to this for more information